### DISCUSSION TITLE
Create event? SSE

### DISCUSSION THREAD

**Discussion by strarsis** [fastapi/fastapi#4654] [Questions]
Does FastAPI have some kind of event/hook system for entity creation?
E.g. a new comment entity is inserted (using POST). A SSE event should be emitted when this happens.

**Comment by havardthom** [comment]
I would recommend using some kind of message broker for this, e.g. Redis pub/sub interface (or maybe your database has some event system?). So connecting to an SSE endpoint will subscribe to a redis pub/sub channel and start listening. When other endpoints are called you can publish message to the channel and the listener on SSE endpoint will emit what you want. Having this system in a separate service outside FastAPI is preferred since you want a single source of truth and FastAPI could run in many processes/containers, so you want it to be stateless.

**Comment by strarsis** [comment]
@havardthom: This makes sense! Is there a message broker that can be easily integrated with FastAPI?
